nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteStreamCommand.java

```diff
@@ -382,10 +389,10 @@ public void onTrigger(ProcessContext context, final ProcessSession session) thro
             Map<String, String> attributes = new HashMap<>();
 
             final StringBuilder strBldr = new StringBuilder();
-            try {
-                String line;
-                while ((line = bufferedReader.readLine()) != null) {
-                    strBldr.append(line).append("\n");
+            try (final InputStream is = new FileInputStream(errorOut)) {
```

Comment:
The java-monitor.com link wouldn't load for me, so I couldn't see your reference. So is this problem here with the BufferedReader? I'm not entirely sure I follow the change here because either way you're pulling data from stderr. This case just redirects it and you're reading it from a file.

Reply:
Strange, the link is not working for me either anymore. There are some pointers on this stackoverflow thread: 
https://stackoverflow.com/questions/16983372/why-does-process-hang-if-the-parent-does-not-consume-stdout-stderr-in-java
The difference is that by writing to a file, you bypass the OS pipe for stderr, and therefore it does not block.
By the way if you execute `TestExecuteStreamCommand` with the old code, you will be able to reproduce the deadlock even with unit tests.